lib/carrierwave/storage/aws_file.rb

```diff
@@ -60,6 +60,10 @@ def copy_to(new_path)
         bucket.object(new_path).copy_from(copy_source: "#{bucket.name}/#{file.key}")
       end
 
+      def signed_url(options = {})
+        uploader.sign_urls[:signer].call(public_url, options)
```

Comment:
I agree with you. My first intention was to have a `config.aws_signer = -> (...){...}` option but unfortunately carrierwave is calling configuration procs as you can see [here](https://github.com/carrierwaveuploader/carrierwave/blob/a1b96d05d021f0ab53c8a2ef460bc96664e52b79/lib/carrierwave/uploader/configuration.rb#L140). 

I'd suggest to add something like
```   
def signer
  uploader.aws_sign_urls.try(:fetch, :signer)
end
```
in order to get things a little bit cleaner. What do you think?